### PR TITLE
Improve Akka.Actor.Cancelable

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes
- Add new Create methods that takes CancellationToken params
- Clean up the Dispose method
- Expose the _isDisposed field as a property

### CODE REVIEW

File: src/core/Akka/Actor/Cancellation/Cancelable.cs [hunk 1]
```diff
@@ -207,27 +206,17 @@ protected virtual void Dispose(bool disposing)
...
+                    _source?.Dispose();
                 }
-                _isDisposed = true;
-            }
-            finally
-            {
-                // base.dispose(disposing);
+                //Clean up unmanaged resources
             }
+            IsDisposed = true;
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Cancellation/Cancelable.cs:219]: Why get rid of the `try...finally` here,  in case `source.Dispose()` throws?

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
API changes look fine, but have questions on the underlying behavior.

File: src/core/Akka/Actor/Cancellation/Cancelable.cs (hunk 1 above)
Comment (Arkatufus) [reply src/core/Akka/Actor/Cancellation/Cancelable.cs:219]: Hmmm... I can't see a CancellationTokenSource to throw inside its Dispose method, but i'll add the try...catch block back

Comment (Arkatufus) [reply src/core/Akka/Actor/Cancellation/Cancelable.cs:219]: To my defense, the original code does not catch any exception thrown from Dispose, it only have an empty finally block

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**